### DISCUSSION TITLE
Lexical Validation vs Schematron Validation

### DISCUSSION THREAD

**Discussion by luispozoWS** [OP-TED/eForms-SDK#851] [Q&A] [answered]
Hello,
Earlier this month we pointed that sometimes we don't have certain schematron rules and notices could be rejected in our application.
We had this answer : "The validation process starts with the lexical validation (i.e. validation against the XSD) and the schematron will only be evaluated after".
What does this imply ? If schematron rules don't cover all possible scenarios what is the buyer supposed to see in the application to avoid sending an incomplete notice?
Best regards

**Comment by YvesJo** [comment, marked as answer]
Hi,
All the CVS applied controls are in the shared SDK, so if it was deliberately decided not to implement some existing controls, some errors could be detected lately.
The lexical validation is about conformance to the XSD to ensure that the correct elements are used at the correct place. If the this is not verified, then the schematron rules, which are XPATH based, could lead to an unexpected behavior. The XSD is relatively permissive and it is important to make sure that the application is generating an XSD conform XML, else the user would face an XPATH based, hence non-user friendly, message. The challenge may exist with some identifiers which are mandated by the schema and should therefore be present when other siblings exist.
The user should make sure he has at least filled in all the unconditionally mandatory fields (schematron stage 2) before submitting
